src/app/api/admin/roles/wizard/templates/route.ts

```
@@ -0,0 +1,185 @@
+ * @param req - The request object
+ * @returns A response with the permission templates
+ */
+export const GET = withACL(
+  withTenant(async (req: NextRequest) => {
+    try {
+      const { searchParams } = new URL(req.url);
+      const scope = searchParams.get('scope') as RoleScope || 'tenant';
+      const siteId = searchParams.get('siteId') || undefined;
+      
+      // Filter templates based on scope
+      const filteredTemplates = PERMISSION_TEMPLATES.filter(template => {
```

ESLint

[error] 'PERMISSION_TEMPLATES' was used before it was defined.

(no-use-before-define)